script.js

```diff
@@ -0,0 +1,136 @@
+// Start here
+
+// Step 1 - Welcome and introduction
+// Your code goes here
+alert(
+  `Welcome to our Javascript Pizzeria. Ready to Start? - Click 'OK' to begin.`
+);
+// Asking for name
+
+  let userName = prompt("What is your name?");
+
+// follow up greeting with name
+
+if (userName) {
+  alert(`Hello ${userName}! Welcome!`);
+} else {
+  alert(`Hello there! Welcome `);
+}
+
+// Step 2 - Food choice
+
+//asking for userChoice
+let chosenPizza;
+
+while (true) {
+
+let userChoice = prompt(`Enter the number of your chosen Pizza: 
+1. Margerita 
+2. Pepperoni 
+3. Hawaiian`);
+
+if (userChoice === "1") {
+  chosenPizza = "Margerita";
+  break;
+} else if (userChoice === "2") {
+  chosenPizza = "Pepperoni";
+  break;
+} else if (userChoice === "3") {
+  chosenPizza = "Hawaiian";
+  break;
+} else {
+  alert("Invalid choise. Please select 1,2 or 3");
+}
+
+}
+alert(`You chose a ${chosenPizza} pizza`);
+
+// Step 3 - Subtype choice
+// Your code goes here
+
+let chosenSize;
+
+while (true) {
+
+let userChoice2 = prompt(`Please select size: 
+1. Small 
+2. Normal
+3. Family-size `);
+
+if (userChoice2 === "1") {
+  chosenSize = "Small";
+  break;
+} else if (userChoice2 === "2") {
+  chosenSize = "Normal";
+  break;
+} else if (userChoice2 === "3") {
+  chosenSize = "Family-size";
+  break;
+} else {
+  alert("Please select a size");
+}
+
+}
+alert(`Thank you for choosing the ${chosenSize} ${chosenPizza} pizza`);
+
+
+
+// Step 4 - Age
+// Asking for age and giving tiered price based on answer
+
+let userAge = prompt("Please enter your age:");
+
+//user input converted into number
+userAge = parseInt(userAge);
+
+// checking that input is Not a "not a number" input
+
+if (!isNaN(userAge)) {
+  // then if that is ok - determined tiered price
+  let price;
+  if (userAge < 19) {
+    price = 10; //Child price
+  } else if (userAge >19 && userAge <=65) { //"&&" instead of "&" because it checks if both conditions are met
+    price = 20; //adult price
+  } else {
+    price = 15; //senior price
+  }
+
+//Adjust the price according to size - late fix  
+let priceAdjust= "";
```

Comment:
this is great and makes sense in the real world for someone to pay more/less for the size of pizza, commenting that it is a late fix is also helpful to show you thought about it